drivers/i2c/i2c_cc32xx.c

```diff
@@ -0,0 +1,417 @@
+/*
+ * Copyright (c) 2017, Texas Instruments Incorporated
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* The logic here is adapted from SimpleLink SDK's I2CCC32XX.c module. */
+
+#include <kernel.h>
+#include <errno.h>
+#include <i2c.h>
+#include <soc.h>
+#include "i2c-priv.h"
+
+/* Driverlib includes */
+#include <inc/hw_memmap.h>
+#include <inc/hw_common_reg.h>
+#include <driverlib/rom.h>
+#include <driverlib/rom_map.h>
+#include <driverlib/i2c.h>
+
+#define I2C_MASTER_CMD_BURST_RECEIVE_START_NACK	 I2C_MASTER_CMD_BURST_SEND_START
+#define I2C_MASTER_CMD_BURST_RECEIVE_STOP \
+	I2C_MASTER_CMD_BURST_RECEIVE_ERROR_STOP
+#define I2C_MASTER_CMD_BURST_RECEIVE_CONT_NACK	 I2C_MASTER_CMD_BURST_SEND_CONT
+
+#define I2C_SEM_MASK \
+	COMMON_REG_I2C_Properties_Register_I2C_Properties_Register_M
+#define I2C_SEM_TAKE \
+	COMMON_REG_I2C_Properties_Register_I2C_Properties_Register_S
+
+#define IS_I2C_MSG_WRITE(flags) ((flags & I2C_MSG_RW_MASK) == I2C_MSG_WRITE)
+
+/* Remove if/when this shows up in i2c.h: */
+#define I2C_SPEED_GET(cfg)		(((cfg) & I2C_SPEED_MASK) \
+						>> I2C_SPEED_SHIFT)
+
+#define DEV_CFG(dev) \
+	((const struct i2c_cc32xx_config *const)(dev)->config->config_info)
+#define DEV_DATA(dev) \
+	((struct i2c_cc32xx_data *const)(dev)->driver_data)
+#define DEV_BASE(dev) \
+	((DEV_CFG(dev))->base)
+
+
+enum i2c_cc32xx_state {
+	/* I2C is idle, and not performing a transaction */
+	I2C_CC32XX_IDLE_MODE = 0,
+	/* I2C is currently performing a write operation */
+	I2C_CC32XX_WRITE_MODE,
+	/* I2C is currently performing a read operation */
+	I2C_CC32XX_READ_MODE,
+	/* I2C error has occurred */
+	I2C_CC32XX_ERROR = 0xFF
+};
+
+struct i2c_cc32xx_config {
+	u32_t base;
+	u32_t bitrate;
+	unsigned int exception_no;
```

Comment:
Maybe we should change it irq_no? More conventional.

Reply:
The TI SDK and device documentation consistently uses interrupt numbers in their documentation and device header files.
Zephyr however uses "exception numbers" (which are interrupt numbers shifted down by 16): eg: see ARM technical documentation: http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0203h/Cihbffg3.html .     
Since this is somewhat confusing when writing code which mixes Zephyr API calls that take exception numbers and TI driverlib APIs which use "interrupt numbers", the variable name here is meant to clarify the true meaning of the value.

Reply:
Hmm, seems it's the other way around: TI uses exception numbers. OK, will change to irq_no.